### DISCUSSION TITLE
Super slow tsserver feedback (compared to VSCode)

### DISCUSSION THREAD

**Discussion by baptisteArno** [zed-industries/zed#8884] [Feature Requests]
Anyone else notices this on big TypeScript project.

It's barely useable to be honest.

For example, I see a type issue, I change it and have to wait for more than 5 seconds to have visual feedback on the editor.

When I compare to VSCode, it's much much faster. But why? They both use tsserver, right?

I'd love to help debug this but I'm not sure how to proceed.

**Comment by Moshyfawn** [comment]
I'm not sure about the solution to your performance problem, but there was a recent discussion in #8821 about spawning separate ts servers to try and negate the problem in larger codebases.

**Comment by JosephTLyons** [comment]
@baptisteArno - I realize this is an older issue, but we are looking to improve things here, but we are struggling to find ways to faithfully reproduce the performance issues here. Do you have any projects that consistently had performance issues that you could share? 

cc: @osiewicz

**Reply by baptisteArno** [reply]
Yes, my project is open-source :)

https://github.com/baptisteArno/typebot.io

Let me know if you don't manage to reproduce the slowliness. In my experience, it is OK when opening the project but at some point it was super slow.

**Comment by stenolmre** [comment]
For me it has happened for larger monorepos, for example using turborepo and nextjs. Without monorepo everything has been working great, even with larger codespaces. Also I would like to point out that when opening the editor, everything is fine but after couple of changes in files, it starts to lag and sometimes closing and opening file will fix it, but not always.

Sorry, I cannot provide any repository since all of them are private.

**Comment by JosephTLyons** [comment]
We are shipping `VTSLS` as the default language server in Zed. I'm curious to hear if anyone here has a better experience using this language server on the same or similar projects as before:

- https://zed.dev/releases/stable/0.141.2

**Reply by stenolmre** [reply]
It seems like the issue is resolved, at least for me. Over the last couple of days I haven't noticed it anymore, after the release. Thanks for the update 👍

**Comment by SebastianCCC** [comment]


**Reply by baptisteArno** [reply]
https://github.com/zed-industries/zed/issues/18698

**Reply by SebastianCCC** [reply]
Thanks 👍

**Comment by typeofweb** [comment]
I still experience the problem. Often type errors take a few seconds to disappear after I’ve fixed the issue. Never experienced that in VSCode.

**Comment by bsmith-self** [comment]
I'm also experiencing this on the latest Zed. It applies not just to errors, but also to things like autocomplete. Just now I typed <object> <dot> and waited several seconds before the properties showed up (on a type with only five properties)

**Reply by PATEL7VISHAL** [reply]
Same as here, zed is really slow for typescript LSP completion, it does not even complete with neovim LSP, though both use the same LSP server. also sometimes the LSP gets stopped working after sometime(after 10-20 min) and need to restart the LSP server(faced in rust code).

I am still preferring VsCode or NeoVim over the Zed. It is disappointing that multiple times I have tried new zed version(current 0.178.2) but this basic LSP things not get resolved yet

**Comment by dohomi** [comment]
I tried my Typescript React project which is quite a large codebase with lots of components with VSCode, Webstorm and now Zed back to back.

The amount of time I wait for any autocomplete suggestion hitting Crgl + Space is very slow comparing Zed to VScode and Webstorm. I disabled 
```
  "features": {
    "edit_prediction_provider": null
  },
  ```
  Also as I have 64GB ram gave the lsp way more ram to work with, nothing speed up the type lookup
  ```
  "lsp": {
    "vtsls": {
      "settings": {
        // For TypeScript:
        "typescript": { "tsserver": { "maxTsServerMemory": 16184 } },
        // For JavaScript:
        "javascript": { "tsserver": { "maxTsServerMemory": 16184 } }
      }
    }
  }
  ```
Its a pretty naked Zed installation with only the default config except the above adjustments. 
Any idea what else I could try? Zed is so fast in general handling and I love the design, just the autocomplete/suggestions of my Typescript files is non-usable.

**Reply by baptisteArno** [reply]
I agree, just unusable on large TS projects. Further discussed here: https://github.com/zed-industries/zed/issues/18698

**Comment by JohnRTitor-Work** [comment]
Still the same issue.

https://github.com/zed-industries/zed/issues/18698

I want to echo that zed is fast in general, but because of this issue, I am having trouble recommending zed to other people in my team.

**Comment by VaheAA** [comment]
I could say that the issue is still actual and its getting slower and slower with every new added file or code piece. Was anybody able at least to make it a bit more alive? 

`Version: 0.217.3`

**Reply by lubiah** [reply]
I'm also experiencing this slowness

**Reply by pridees** [reply]
The same issue. I wait 2-3-5 sec for each edit to see re-evaluation types and errors on my TS codebase (100k loc). VSCode is much faster. No idea what is actual Zed's advantage over competitors.

**Reply by youssef-lr** [reply]
I'm seeing this slowness even with `tsgo` in neovim. I think VSCode does some optimization under the hood.

**Reply by VaheAA** [reply]
Hmm, so this is probably something with language server itself? Makes sense yes.

**Reply by PATEL7VISHAL** [reply]
> I'm seeing this slowness even with `tsgo` in neovim. I think VSCode does some optimization under the hood.

but `ts_server` on neovim works smoother and faster then zed, I guess both neovim and zed using same lsp typescript-language-server but neovim works faster, also vtsls also work faster then zed, so definitively some problem in zed lsp impl.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by sheikhameen** [comment]
I've been trying to debug the sluggishness on zed for some time now, specifically ctrl+space for suggestions. And then I decided to go back to vscode to make sure if zed is actually slow. Turns out, in my case, both vscode and zed take similar delay to give any meaningful feedback like autocomplete suggestions. What made me feel vscode was faster is the little loading indicator as soon as I type ctrl+space. Apart from that they both take roughly the same delay. Nonetheless, both of them are slow to me.

**Reply by VaheAA** [reply]
This is language server issue i think, so as long as lang server is slow, editors will suffer

**Reply by youssef-lr** [reply]
How can it be a language server issue if it's pretty fast in VSCode?

**Comment by Aliireza1101** [comment]
I've been trying to choose one editor between vs code and zed, i set my laptop usage to battery saver (to feel the difference) and tried writing some typescript code in the both vs code and zed, vs code was giving faster feedback and autocomplete than zed does, but when I open zed, it loads way faster than vscode
